Installation docu (#1304)
kyma-project/kyma: docs/kyma/docs/017-details-local-installation.md

---
title: Local installation
type: Details
---

For better understanding of complex installation process, this guide describes it step by step.

> 

Author: first one already adressed

> 

Author: the rest addressed

> ## Start local installation

Reviewer: `## Start the local installation`

> To fire local installation run following command:

Author: done

> To fire local installation run following command:

Author: done

```
./installation/cmd/run.sh
```

> This script sets up default parameters, starts Minikube, builds Kyma-Installer, generates local configuration, creates `installer` Custom Resource and sets up Installer. Subsequent sections provide a detailed description of each step.

Reviewer: - Kyma-Installer -> and not `the Kyma installer` like here: https://github.com/kyma-project/kyma/blob/96313927b69dda5921a6024f390dc697b56eddf8/installation/README.md ?
- creates `installer` Custom Resource and sets up Installer. -> ``creates the `installation` Custom Resource, and sets up the Installer. ``  -  isn't the CR called `installation` like here: https://github.com/kyma-project/kyma/blob/master/docs/kyma/docs/040-cr-installation.md ?

Author: changed

Author: changed

### Installation parameters

You can execute the `installation/cmd/run.sh` script with the following parameters:

> - `--skip-minikube-start` - it skips execution of the `installation/scripts/minikube.sh` script. See the "Start Minikube" section for more details.

Reviewer: - `--skip-minikube-start` - it skips -> - `` `--skip-minikube-start` which skips`` (guidelines)
- "Start Minikube" ->` **Start Minikube** ` (guidelines)

> - `--vm-driver` -  either `virtualbox` or `hiperkit` depending on your operating system

Reviewer: -  `--vm-driver` -  either `virtualbox` or `hiperkit` depending on your operating system -> `` `--vm-driver` points to  either `virtualbox` or `hiperkit` depending on your operating system.`` (let's be consistent and add a verb here + full sentence so a period at the end)

> ## Start Minikube

Reviewer: `## The minikube.sh script`?

> **NOTE:** To work with Kyma, use only the provided scripts and commands. Kyma does not work on a basic Minikube cluster that you can start using the `minikube start` command or stop with the `minikube stop` command. If you don't need Kyma on Minikube anymore, remove the cluster with the `minikube delete` command.

> Purpose of `installation/scripts/minikube.sh` is to configure and start Minikube. The script also checks if your development environment is cofigured to handle Kyma installation. This includes checking Minikube and Kubectl versions. In case Minikube is already initialized you will be prompted to agree to remove previous Minikube cluster. Script exits if you don't want to restart your cluster.

Reviewer: > The purpose of `installation/scripts/minikube.sh` is to configure and start Minikube. The script also checks if your development environment is configured to handle the Kyma installation. This includes checking Minikube and kubectl versions. If Minikube is already initialized, the system prompts you to agree to remove the previous Minikube cluster. The script exits if you don't want to restart your cluster.

> >**NOTE:** Minikube is configured to disable default nginx ingress controller.

Reviewer: ` Minikube is configured to disable the default Nginx Ingress Controller.` (no need to add a note, + there is one below already)

> >**NOTE:** For the complete list of parameters passed to `minikube start` command, refer to the `installation/scripts/minikube.sh` script.

Reviewer: to `minikube start` command, ->`` to the `minikube start` command,``

> Once Minikube is up and running, the script adds local installation entries to /etc/hosts.

Reviewer: /etc/hosts. -> `` `/etc/hosts`.``

> ## Build Kyma-Installer

Reviewer: ` ##  The build-kyma-installer.sh script`

> Installer is application based on a [Kubernetes operator](https://coreos.com/operators/). Its purpose is to install Helm charts defined in the Installer Custom Resource. Kyma-Installer is a Docker image that bundles Installer binary with Kyma charts. 

Reviewer: > Installer is an application based on a [Kubernetes Operator](https://coreos.com/operators/). Its purpose is to install Helm charts defined in the `Installation` Custom Resource. The Kyma installer is a Docker image that bundles the Installer binary with Kyma charts. 

- the `Installation` Custom Resource, right? If anything else, pls adjust it in the whole text to be consistent.
- the Kyma installer, right?  If anything else, pls adjust it in the whole text to be consistent.

> `installation/scripts/build-kyma-installer.sh` script extracts Kyma-Installer image name from the `installer.yaml` deployment file and uses it to build a Docker image inside Minikube. This image will contain local Kyma sources from `resources` folder. 

Reviewer: - `installation/scripts/build-kyma-installer.sh` script -> `` The `installation/scripts/build-kyma-installer.sh` script``
- Kyma-Installer image -> the Kyma installer image 
- This image will contain local Kyma sources from `resources` folder.  -> ``This image contains local Kyma sources from the `resources` folder. ``

> >**NOTE:** For Kyma-Installer Docker image details refer to the `kyma-installer/kyma.Dockerfile` file.

Reviewer: For Kyma-Installer Docker image details refer  -> `For the Kyma installer Docker image details, refer `

## Generate local configuration for Azure-Broker

> 

Reviewer: Could you pls add a sentence that explains how this section relates to the whole idea of the doc?

The Azure-Broker sub-component is a part of the `core` deployment that provisions managed services in the Microsoft Azure cloud. To enable Azure-Broker, export the following environment variables:
 - AZURE_BROKER_SUBSCRIPTION_ID
 - AZURE_BROKER_TENANT_ID
 - AZURE_BROKER_CLIENT_ID
 - AZURE_BROKER_CLIENT_SECRET

>**NOTE:** As the Azure credentials are converted to a Kubernetes Secret, make sure the exported values are base64-encoded.

> ## Create installer custom resource

Reviewer: `## The create-cr.sh script`?

> `installation/scripts/create-cr.sh` script prepares Installer Custom Resource from `installation/resources/installer-cr.yaml.tpl` template. In local installation scenario default Installer Custom Resource will be used. `kyma-installer` already contains local Kyma resources bundled, thus `url` is ignored by Installer component. 

Reviewer: - `installation/scripts/create-cr.sh` script -> `` The `installation/scripts/create-cr.sh` script ``
- Installer Custom Resource -> ``the `installation` Custom Resource`` ?
- from -> from the
- In local installation scenario default Installer Custom Resource will be used.  ->`` The local installation scenario uses the default `installation` Custom Resource.`` (guidelines - > avoid passive voice and future tenses)
- Installer component. -> the Installer component.

> >**NOTE:** For Installer Custom Resource details refer to the `docs/kyma/docs/040-cr-installation.md` file.

Reviewer: ``>**NOTE:** For the `installation` Custom Resource details, refer to the `docs/kyma/docs/040-cr-installation.md` file.``

> ## Kyma-Installer deployment

Reviewer: `## The installer.sh script`?

> The `installation/scripts/installer.sh` script creates the default RBAC Role, installs [Tiller] (https://docs.helm.sh/), and deploys the Kyma-Installer component.

Reviewer: - [Tiller] (https://docs.helm.sh/) ->` [Tiller](https://docs.helm.sh/)` 
- Kyma-Installer -> `the Kyma installer`?

> >**NOTE:** For Kyma-Installer deployment details refer to the `installation/resources/installer.yaml` file.

Reviewer: ``>**NOTE:** For the Kyma installer deployment details, refer to the `installation/resources/installer.yaml` file. ``

> The script applies the Installer Custom Resource and marks it with the `action=install` Label, which triggers the Kyma installation.

Reviewer: - Installer Custom Resource -> ``the `installation` Custom Resource``
- Label -> label

> 

Reviewer: Ok, in general I would put in this doc only explanations of scripts that are triggered by minikube.sh.

> >**NOTE:** Kyma installation runs in background. Execute `./installation/scripts/is-installed.sh` to follow the installation process.

Reviewer: in background -> in the background

---
title: Local installation
type: Details
---

For better understanding of complex installation process, this guide describes it step by step.

## Start local installation

To fire local installation run following command:
```
./installation/cmd/run.sh
```

This script sets up default parameters, starts Minikube, builds Kyma-Installer, generates local configuration, creates `installer` Custom Resource and sets up Installer. Subsequent sections provide a detailed description of each step.

### Installation parameters

You can execute the `installation/cmd/run.sh` script with the following parameters:

- `--skip-minikube-start` - it skips execution of the `installation/scripts/minikube.sh` script. See the "Start Minikube" section for more details.
- `--vm-driver` -  either `virtualbox` or `hiperkit` depending on your operating system

## Start Minikube

> > **NOTE:** To work with Kyma, use only the provided scripts and commands. Kyma does not work on a basic Minikube cluster that you can start using the `minikube start` command or stop with the `minikube stop` command. If you don't need Kyma on Minikube anymore, remove the cluster with the `minikube delete` command.

Reviewer 2: ```suggestion
> **NOTE:** To work with Kyma, use only the provided scripts and commands. Kyma does not work on a default Minikube cluster started with `minikube start` command. If you don't need Kyma on Minikube anymore, remove the cluster with the `minikube delete` command.
```

Author: this one is note copied from docs/kyma/docs/031-gs local-installation.md

Purpose of `installation/scripts/minikube.sh` is to configure and start Minikube. The script also checks if your development environment is cofigured to handle Kyma installation. This includes checking Minikube and Kubectl versions. In case Minikube is already initialized you will be prompted to agree to remove previous Minikube cluster. Script exits if you don't want to restart your cluster.

>**NOTE:** Minikube is configured to disable default nginx ingress controller.

>**NOTE:** For the complete list of parameters passed to `minikube start` command, refer to the `installation/scripts/minikube.sh` script.

> Once Minikube is up and running, the script adds develop domains to /ets/hosts.

Reviewer 2: ```suggestion
Once Minikube is up and running, the script adds local installation entries to /etc/hosts.
```

## Build kyma-installer

> Kyma installs locally using a proprietary installer based on a [Kubernetes operator](https://coreos.com/operators/).

Reviewer 3: Reword this paragraph - it is not clear.

Author: i removed this sentence completely

> Installer is application based on a [Kubernetes operator](https://coreos.com/operators/). Its purpose is to install Helm charts defined in the Installer Custom Resource. Kyma-Installer is Installer with bundled Kyma charts. 

Reviewer 2: ```suggestion
Installer is application based on a [Kubernetes operator](https://coreos.com/operators/). Its purpose is to install Helm charts defined in the Installer Custom Resource. Kyma-Installer is a Docker image that bundles Installer binary with Kyma charts. 
```

`installation/scripts/build-kyma-installer.sh` script extracts `kyma-installer` image name from the `installer.yaml` deployment file and uses it to build a Docker image inside minikube. This image will contain local kyma sources from `resources` folder. 

>**NOTE:** For `kyma-installer` Docker image details refer to the `kyma-installer/kyma.Dockerfile` file.

## Generate local configuration for Azure-Broker

> The `Azure-Broker` sub-component is a part of the `core` deployment that provisions managed services in the Microsoft Azure cloud. To enable `Azure-Broker`, export the following environment variables:

Reviewer 3: We don't need azure broker any more in the installation guide.

---
title: Local installation
type: Details
---

This document explains process of local installation. For better understanding of complex installation process, this guide describes it step by step.

## Start local installation

To fire local installation run following command:
```
./installation/cmd/run.sh
```

> This script sets up default parameters, starts minikube, builds kyma-installer, generates local configuration, creates installer custom resource and finally sets up installer. Each step is explained in details below.

Author: done

> ### Installation run parameters

Author: done

> Script `installation/cmd/run.sh` can be run with following option:

Author: done

> - `--skip-minikube-start` - it skips execution of the `installation/scripts/minikube.sh` script, for details see section "Start minikube" in this document

Author: done

> - `--cr` - path to the installer custom resource file, when not provided script generates default local custom resource

Reviewer 4: --cr flag will no longer be supported, please remove.

Author: done

## Start minikube

> **NOTE:** To work with Kyma, use only the provided scripts and commands. Kyma does not work on a basic Minikube cluster that you can start using the `minikube start` command or stop with the `minikube stop` command. If you don't need Kyma on Minikube anymore, remove the cluster with the `minikube delete` command.

Purpose of `installation/scripts/minikube.sh` is to configure and start minikube. The script also checks if your development environment is cofigured to handle kyma installation. This includes checking minikube and kubectl versions. In case minikube is already initialized you will be prompted to agree to remove previous minikube cluster. Script exits if you don't want to restart your cluster.

>**NOTE:** Minikube is configured to disable default nginx ingress controller.

>**NOTE:** For the complete list of parameters passed to `minikube start` command, refer to the `installation/scripts/minikube.sh` script.

Once minikube is up and running, the script adds develop domains to /ets/hosts.

## Build kyma-installer

Kyma installs locally using a proprietary installer based on a [Kubernetes operator](https://coreos.com/operators/).

> `installer` is application based on a [Kubernetes operator](https://coreos.com/operators/). Its purpose is to install helm charts defined via installer custom resource. `kyma-installer` is `installer` with bundled kyma charts. 

Reviewer 4: ...defined in the installer Custom Resource..

Author: done

> `installation/scripts/build-kyma-installer.sh` script extracts `kyma-installer` image name from the `installer.yaml` deployment file and use it to build docker image inside minikube. This image will contain local kyma sources from `resources` folder. 

Reviewer 4: ...uses it to build a Docker image...

Author: done

>**NOTE:** For `kyma-installer` docker image details refer to the `kyma-installer/kyma.Dockerfile` file.

> ## Generate local configuration

Reviewer 4: Generate local configuration for Azure-Broker

Author: done

> `installation/scripts/generate-local-config.sh` prepares configuration for optional `azure-broker` sub-component of `core` component. To enable `azure-broker` sub-component you need to prepare following variables along with their values encoded as base64 strings: `AZURE_BROKER_SUBSCRIPTION_ID`, `AZURE_BROKER_TENANT_ID`, `AZURE_BROKER_CLIENT_ID`, and `AZURE_BROKER_CLIENT_SECRET`.

Reviewer 4: The 'Azure-Broker' sub-component is a part of the 'core' deployment that provisions managed services in the Microsoft Azure cloud. To enable 'Azure-Broker', export the following environment variables:
- `AZURE_BROKER_SUBSCRIPTION_ID`
- `AZURE_BROKER_TENANT_ID`
- `AZURE_BROKER_CLIENT_ID`
- `AZURE_BROKER_CLIENT_SECRET`

>**NOTE:** As the Azure credentials are converted to a Kubernetes Secret, make sure the exported values are base64-encoded.

Author: done

```
$ export AZURE_BROKER_SUBSCRIPTION_ID="..."
$ export AZURE_BROKER_TENANT_ID="..."
$ export AZURE_BROKER_CLIENT_ID="..."
$ export AZURE_BROKER_CLIENT_SECRET="..."
```

> The script creates secret out of provided values and enables `azure-broker` sub-component.

Reviewer 4: Remove lines 52-58.

Author: done

## Create installer custom resource

> `installation/scripts/create-cr.sh` script prepares `installer` custom resource from `installation/resources/installer-cr.yaml.tpl` template. In local installation scenario default `installer` custom resource will be used. `kyma-installer` already contains local `kyma` resources budled thus `url` is ignored by `installer` component. 

Reviewer 4: typo in "bundled"

Author: done

>**NOTE:** For `installer` custom resource details refer to the `docs/kyma/docs/040-cr-installation.md` file.

## Kyma-Installer deployment

> `installation/scripts/installer.sh` script deploys `kyma-installer` and start installation. The script creates default rbac role and installs [tiller](https://docs.helm.sh/) as prerequisities for `installer` component. Next installs `kyma-installer`.

Reviewer 4: The `installation/scripts/installer.sh` script creates the default RBAC role, installs [Tiller] (https://docs.helm.sh/), and deploys the `kyma-installer`component.

Author: done

>**NOTE:** For `kyma-installer` deployment details refer to the `installation/resources/installer.yaml` file.

> The script applies `installer` custom resource and triggers kyma installation labeling it with `action=install`.

Reviewer 4: The script applies the `installer` Custom Resource and marks it with the`action=install` label, which triggers the Kyma installation.

Author: done

> >**NOTE:** `kyma` installation runs in background. Execute `./installation/scripts/is-installed.sh` to observe installation.

Reviewer 4: ~~to observe installation~~
to follow the installation process.

Author: done

---
title: Local installation
type: Details
---

> Simple as it may seem, the Kyma installation is a complex and dynamic process. This document extends the “031-gs-local-installation.md” guide with a detailed breakdown of the alternative installation method, i.e. the “run.sh” script.

Reviewer: ``Simple as it may seem, the Kyma installation is a complex and dynamic process. This document extends the **Local Kyma installation** guide with a detailed breakdown of the alternative installation method which is the `run.sh` script.`` 

-> why I actually made those changes: we bold doc titles + as the `kyma/docs` are visible in the UI, we rather refer to them by their titles + script formatting.

Author: done

> > **NOTE:** the ‘run.sh’ script should be used exclusively for development purposes.

Author: done

To start the local installation, run the following command:
```
./installation/cmd/run.sh
```

> This script sets up default parameters, starts Minikube, builds Kyma-Installer, generates local configuration, creates the `installation` Custom Resource and sets up Installer. Subsequent sections provide a detailed description of each step.

Reviewer: - `the Kyma-Installer` -> `the` is missing here
- We have decided within the TW team that we will format the resources this way: `the Installation custom resource` -> would that be ok to update that in the whole doc? Apologies for a different approach during the first review...
- `resource, and sets up`-> serial comma was missing

Author: shouldn't it be `the Installation Custom Resource`? The rest was corrected

You can execute the `installation/cmd/run.sh` script with the following parameters:

> - `--skip-minikube-start` which skips execution of the `installation/scripts/minikube.sh` script. See the **Start Minikube** section for more details

Reviewer: period at the end?

Author: done

> - `--vm-driver` points to either `virtualbox` or `hiperkit` depending on your operating system

Reviewer: should be `which points` (my fault, missed that consistency during the first review) + period at the end?

Author: done

The following snippet lies in the heart of the “run.sh” script:

> 

Reviewer: - Sounds really nice but more for a poem than a tech doc :) Maybe let's keep it simpler here and write sth like `snippet is the main element of`
- “run.sh” -> `` `run.sh` ``

Author: done

```
if [[ ! ${SKIP_MINIKUBE_START} ]]; then
    bash ${CURRENT_DIR}/../scripts/minikube.sh --domain "${DOMAIN}" --vm-driver "${VM_DRIVER}"
fi

bash ${CURRENT_DIR}/../scripts/build-kyma-installer.sh --vm-driver "${VM_DRIVER}"

bash ${CURRENT_DIR}/../scripts/generate-local-config.sh

if [ -z "$CR_PATH" ]; then

    TMPDIR=`mktemp -d "${CURRENT_DIR}/../../temp-XXXXXXXXXX"`
    CR_PATH="${TMPDIR}/installer-cr-local.yaml"

    bash ${CURRENT_DIR}/../scripts/create-cr.sh --output "${CR_PATH}" --domain "${DOMAIN}"
    bash ${CURRENT_DIR}/../scripts/installer.sh --local --cr "${CR_PATH}"

    rm -rf $TMPDIR
else
    bash ${CURRENT_DIR}/../scripts/installer.sh --cr "${CR_PATH}"
fi
```

The subsequent paragraphs describe respective sub-scripts triggered during the installation process.

## The minikube.sh script

> **NOTE:** To work with Kyma, use only the provided scripts and commands. Kyma does not work on a basic Minikube cluster that you can start using the `minikube start` command or stop with the `minikube stop` command. If you don't need Kyma on Minikube anymore, remove the cluster with the `minikube delete` command.

> Purpose of `installation/scripts/minikube.sh` is to configure and start Minikube. The script also checks if your development environment is cofigured to handle Kyma installation. This includes checking Minikube and Kubectl versions. If Minikube is already initialized, the system prompts you to agree to remove the previous Minikube cluster. The script exits if you don't want to restart your cluster.

Reviewer: - Purpose -> The purpose
- Kyma installation -> the Kyma installation
- Kubectl -> kubectl

Author: done

> Minikube is configured to disable default nginx ingress controller.

Reviewer: the default Nginx Ingress Controller

Author: done

>**NOTE:** For the complete list of parameters passed to the `minikube start` command, refer to the `installation/scripts/minikube.sh` script.

Once Minikube is up and running, the script adds local installation entries to `/etc/hosts`.

## The build-kyma-installer.sh script

> Installer is application based on a [Kubernetes operator](https://coreos.com/operators/). Its purpose is to install Helm charts defined in the `installation` Custom Resource. Kyma-Installer is a Docker image that bundles Installer binary with Kyma charts. 

Reviewer: - an application

Author: done

> The `installation/scripts/build-kyma-installer.sh` script extracts Kyma-Installer image name from the `installer.yaml` deployment file and uses it to build a Docker image inside Minikube. This image contains local Kyma sources from `resources` folder. 

Reviewer: ``the `resources` folder``

Author: done

> >**NOTE:** For Kyma-Installer Docker image details, refer to the `kyma-installer/kyma.Dockerfile` file.

Reviewer: the Kyma-Installer

Author: done

## The generate-local-config.sh script

The Azure-Broker sub-component is a part of the `core` deployment that provisions managed services in the Microsoft Azure cloud. To enable Azure-Broker, export the following environment variables:
 - AZURE_BROKER_SUBSCRIPTION_ID
 - AZURE_BROKER_TENANT_ID
 - AZURE_BROKER_CLIENT_ID
 - AZURE_BROKER_CLIENT_SECRET

> 

Reviewer: ...and that is done irrespective of the `run.sh` script / as part of it / additionally?

>**NOTE:** As the Azure credentials are converted to a Kubernetes Secret, make sure the exported values are base64-encoded.

## The create-cr.sh script

> The `installation/scripts/create-cr.sh` script prepares the `installation` Custom Resource from the `installation/resources/installer-cr.yaml.tpl` template. The local installation scenario uses the default `installation` Custom Resource. Kyma-Installer already contains local Kyma resources bundled, thus `url` is ignored by the Installer component. 

Reviewer: the Kyma-Installer ? (here and there?)

Author: i guess done (here, cause idk where there is).

> >**NOTE:** For the `installation` Custom Resource details, refer to the `docs/kyma/docs/040-cr-installation.md` file.

Reviewer: and again - my bad, Doc title that is displayed should be here so: `the **Installation** document.`

Author: done

## The installer.sh script

The `installation/scripts/installer.sh` script creates the default RBAC Role, installs [Tiller](https://docs.helm.sh/), and deploys the Kyma-Installer component.

> >**NOTE:** For Kyma-Installer deployment details, refer to the `installation/resources/installer.yaml` file.

Reviewer: the Kyma-Installer

Author: done

> ---

Reviewer: 017-details-local-installation-scripts.md?

Author: done

    CR_PATH="${TMPDIR}/installer-cr-local.yaml"

    bash ${CURRENT_DIR}/../scripts/create-cr.sh --output "${CR_PATH}" --domain "${DOMAIN}"
    bash ${CURRENT_DIR}/../scripts/installer.sh --local --cr "${CR_PATH}"

    rm -rf $TMPDIR
else
    bash ${CURRENT_DIR}/../scripts/installer.sh --cr "${CR_PATH}"
fi
```

The subsequent paragraphs describe respective sub-scripts triggered during the installation process.

## The minikube.sh script

> **NOTE:** To work with Kyma, use only the provided scripts and commands. Kyma does not work on a basic Minikube cluster that you can start using the `minikube start` command or stop with the `minikube stop` command. If you don't need Kyma on Minikube anymore, remove the cluster with the `minikube delete` command.

The purpose of `installation/scripts/minikube.sh` is to configure and start Minikube. The script also checks if your development environment is cofigured to handle the Kyma installation. This includes checking Minikube and kubectl versions. If Minikube is already initialized, the system prompts you to agree to remove the previous Minikube cluster. The script exits if you don't want to restart your cluster.

> Minikube is configured to disable the default nginx ingress controller.

Reviewer: the default Nginx Ingress Controller

Author: done

>**NOTE:** For the complete list of parameters passed to the `minikube start` command, refer to the `installation/scripts/minikube.sh` script.

Once Minikube is up and running, the script adds local installation entries to `/etc/hosts`.

## The build-kyma-installer.sh script

Installer is an application based on a [Kubernetes operator](https://coreos.com/operators/). Its purpose is to install Helm charts defined in the Installation custom resource. Kyma-Installer is a Docker image that bundles Installer binary with Kyma charts. 

The `installation/scripts/build-kyma-installer.sh` script extracts Kyma-Installer image name from the `installer.yaml` deployment file and uses it to build a Docker image inside Minikube. This image contains local Kyma sources from the `resources` folder. 

>**NOTE:** For the Kyma-Installer Docker image details, refer to the `kyma-installer/kyma.Dockerfile` file.

## The generate-local-config.sh script

> The `generate-local-config.sh` script configures optional sub-components. At the moment only Azure-Broker is optional sub-component of the `core` deployment. 

Reviewer: At the moment, only Azure-Broker is an optional

Author: done

The Azure-Broker sub-component is a part of the `core` deployment that provisions managed services in the Microsoft Azure cloud. To enable Azure-Broker, export the following environment variables:
 - AZURE_BROKER_SUBSCRIPTION_ID
 - AZURE_BROKER_TENANT_ID
 - AZURE_BROKER_CLIENT_ID
 - AZURE_BROKER_CLIENT_SECRET

> >**NOTE:** You need to exportabove environment variable before executing the `installation/cmd/run.sh` script. As the Azure credentials are converted to a Kubernetes Secret, make sure the exported values are base64-encoded.

Reviewer: exportabove environment variable -> export above environment variables

Author: done
